File: apps/desktop/src/types/sync.ts
import type { User } from "@/services/user";

export type {
  ApiResponse,
  VersionInfo,
  WebsiteGroupDto,
  WebsitesDto,
  SearchEngineDto,
  SyncDataDto,
  ServerSyncData,
  ClientSyncPayload,
  StartSyncResponse,
  ClientSyncDataChunk,
  ClientInfoDto,
  CurrentUserPayload,
  ClientCompatibilityRequest,
  CompatibilityInfo,
} from "@navix/shared-ts";

export { DataType, dataTypeSchema } from "@navix/shared-ts";
export { APP_ERROR_CODES } from "@navix/shared-ts";

export interface SyncStatusUpdaters {
  setIsSyncing: (isSyncing: boolean) => void;
  setSyncMessage: (message: string) => void;
  setSyncCompleted: (completed: boolean) => void;
  incrementDataVersion: () => void;
  switchActiveUser: (user: User) => void;
  refreshAvailableUsers: () => Promise<User[]>;
  t: (key: string, options?: Record<string, unknown>) => string;
}

File: apps/server/src/models/sync.rs
// 兼容历史模块路径：server 内部继续使用 `crate::models::sync::*`，
// 实际类型定义统一来自 shared-rs，避免 server/client DTO 漂移。
pub use shared_rs::dto::sync::{
    ClientInfoDto, ClientSyncData, ClientSyncDataChunk, ClientSyncPayload, CompleteSyncPayload,
    DataType, ServerSyncData, StartSyncResponse, SyncDataDto,
};
